src/init/mod.rs

```
@@ -0,0 +1,290 @@
+use anyhow::Result;
+use inquire;
+use itertools::Itertools;
+use minijinja;
+use regex::Regex;
+use serde;
+use std::collections::HashMap;
+use std::fmt;
+use tracing::{debug, error, info, trace, warn};
+
+use crate::configparser::config;
+use crate::utils::render_strict;
+
+pub mod example_values;
+pub mod templates;
+
+pub fn interactive_init() -> inquire::error::InquireResult<config::RcdsConfig> {
+    println!("For all prompts below, press Enter to leave blank.");
+    println!("All fields that can be set in rcds.yaml can also be set via environment variables.");
+    println!("See the docs for more info on each field: https://beavercds.info/reference/rcds-yaml-reference.html");
+
+    let difficulty_names; // set during `points` prompt later
+
+    // FORMATTING NOTE: Some of these help messages cause rustfmt to silently
+    // fail to format this struct definition. Commenting out the marked
+    // help_message lines temporarily will let the formatting work.
+    //
+    // Ref:
+    // - https://github.com/rust-lang/rustfmt/issues/6687
+    // - https://github.com/rust-lang/rustfmt/issues/3863
```

Rustfmt currently has some issues with formatting this file due to line length limitations. The issues linked here have more context (https://github.com/rust-lang/rustfmt/issues/6687, https://github.com/rust-lang/rustfmt/issues/3863) -- TLDR rustfmt silently fails on really long lines/method chains and thus fails to  format this file unless the lines marked with `// too long to format` are commented out.

@cacama-valvata and I discussed offline a while ago about this and he recommended that this is is documented in the Discord rather than a code comment here since this seems unprofessional. I disagree with that point and prefer to have it documented here so that it cannot get lost or forgotten and have to figure this all out again in the future, because this is a bug in rustfmt and not (personally) a code smell.

Do any of the rest of you have thoughts on this either way?

---

TIL this is a problem. I wonder how often it's happened to me and I never noticed before?

I tried `format_strings` with the nightly toolchain and it doesn't help here. I don't think it's worth breaking the strings manually, either. I think the status quo is fine even if the `too long to format` comments aren't  necessary.